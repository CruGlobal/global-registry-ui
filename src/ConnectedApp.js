import React from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'
import { createBrowserHistory } from 'history'
import { createStore } from './store'
import App from './App'

class ConnectedApp extends React.Component {
  constructor (props) {
    super(props)
    this.history = createBrowserHistory()
    this.store = createStore(this.history)
  }

  render () {
    return (
      <Provider store={this.store}>
        <ConnectedRouter history={this.history}>
          <App/>
        </ConnectedRouter>
      </Provider>
    )
  }
}

export default ConnectedApp
