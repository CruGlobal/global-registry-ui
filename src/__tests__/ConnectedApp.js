import React from 'react'
import ConnectedApp from '../ConnectedApp'
import { render } from 'react-testing-library'

describe('ConnectedApp', () => {
  it('should render without error', () => {
    const {container} = render(<ConnectedApp/>)
    expect(container).toMatchSnapshot()
  })
})
