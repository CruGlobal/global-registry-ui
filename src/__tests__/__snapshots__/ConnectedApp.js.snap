// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedApp should render without error 1`] = `
<div>
  <div
    class="App-root-1"
  >
    <header
      class="MuiPaper-root-23 MuiPaper-elevation4-29 MuiAppBar-root-14 MuiAppBar-positionAbsolute-16 MuiAppBar-colorPrimary-21 AppFrame-appBar-2"
    >
      <div
        class="MuiToolbar-root-50 MuiToolbar-regular-52 AppFrame-toolbar-4"
      >
        <button
          aria-label="Open drawer"
          class="MuiButtonBase-root-60 MuiIconButton-root-54 MuiIconButton-colorInherit-55 AppFrame-menuButton-6"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-59"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-63"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root-187"
          />
        </button>
        <h1
          class="MuiTypography-root-72 MuiTypography-h6-89 MuiTypography-colorInherit-101 MuiTypography-noWrap-98 AppFrame-title-8"
        >
          OpenAPI Specification
        </h1>
        <div
          class="MuiPaper-root-23 MuiPaper-elevation0-25 MuiPaper-rounded-24 AccessTokenInput-paper-111"
        >
          <div
            class="MuiInputBase-root-207 MuiInput-root-194 AccessTokenInput-inputRoot-112 MuiInputBase-focused-209 MuiInput-focused-196"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-217 MuiInput-input-202 AccessTokenInput-input-113"
              placeholder="Access Token"
              type="text"
              value=""
            />
          </div>
          <hr
            class="MuiDivider-root-157 AccessTokenInput-divider-117"
          />
          <button
            class="MuiButtonBase-root-60 MuiIconButton-root-54 MuiIconButton-colorInherit-55 AccessTokenInput-iconDone-114"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-59"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-63"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root-187"
            />
          </button>
          <button
            class="MuiButtonBase-root-60 MuiIconButton-root-54 MuiIconButton-colorInherit-55 AccessTokenInput-iconCancel-115"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-59"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-63"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root-187"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      class="MuiDrawer-root-145 MuiDrawer-docked-146"
    >
      <div
        class="MuiPaper-root-23 MuiPaper-elevation0-25 MuiDrawer-paper-147 AppFrame-drawerPaper-9 AppFrame-drawerPaperClose-10 MuiDrawer-paperAnchorLeft-148 MuiDrawer-paperAnchorDockedLeft-152"
      >
        <div
          class="AppFrame-toolbarIcon-5"
        >
          <button
            class="MuiButtonBase-root-60 MuiIconButton-root-54"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-59"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-63"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root-187"
            />
          </button>
        </div>
        <hr
          class="MuiDivider-root-157"
        />
        <ul
          class="MuiList-root-163 MuiList-padding-164"
          data-testid="navigation"
        >
          <a
            aria-current="page"
            class="MuiButtonBase-root-60 MuiListItem-root-167 MuiListItem-default-170 MuiListItem-gutters-175 MuiListItem-button-176 Navigation-active-162"
            data-testid="navlink-/"
            href="/"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root-179"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-63"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root-180"
            >
              <span
                class="MuiTypography-root-72 MuiTypography-body1-81 MuiListItemText-primary-183"
              >
                OpenAPI Spec
              </span>
            </div>
            <span
              class="MuiTouchRipple-root-187"
            />
          </a>
          <a
            class="MuiButtonBase-root-60 MuiListItem-root-167 MuiListItem-default-170 MuiListItem-gutters-175 MuiListItem-button-176"
            data-testid="navlink-/entity_types"
            href="/entity_types"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root-179"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-63"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <defs>
                  <path
                    d="M0 0h24v24H0z"
                    id="a"
                  />
                </defs>
                <path
                  clip-rule="evenodd"
                  d="M13 9.5h5v-2h-5v2zm0 7h5v-2h-5v2zm6 4.5H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h14c1.1 0 2 .9 2 2v14c0 1.1-.9 2-2 2zM6 11h5V6H6v5zm1-4h3v3H7V7zM6 18h5v-5H6v5zm1-4h3v3H7v-3z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root-180"
            >
              <span
                class="MuiTypography-root-72 MuiTypography-body1-81 MuiListItemText-primary-183"
              >
                Entity Types
              </span>
            </div>
            <span
              class="MuiTouchRipple-root-187"
            />
          </a>
          <a
            class="MuiButtonBase-root-60 MuiListItem-root-167 MuiListItem-default-170 MuiListItem-gutters-175 MuiListItem-button-176"
            data-testid="navlink-/systems"
            href="/systems"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root-179"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-63"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.81 4.47c-.08 0-.16-.02-.23-.06C15.66 3.42 14 3 12.01 3c-1.98 0-3.86.47-5.57 1.41-.24.13-.54.04-.68-.2-.13-.24-.04-.55.2-.68C7.82 2.52 9.86 2 12.01 2c2.13 0 3.99.47 6.03 1.52.25.13.34.43.21.67-.09.18-.26.28-.44.28zM3.5 9.72c-.1 0-.2-.03-.29-.09-.23-.16-.28-.47-.12-.7.99-1.4 2.25-2.5 3.75-3.27C9.98 4.04 14 4.03 17.15 5.65c1.5.77 2.76 1.86 3.75 3.25.16.22.11.54-.12.7-.23.16-.54.11-.7-.12-.9-1.26-2.04-2.25-3.39-2.94-2.87-1.47-6.54-1.47-9.4.01-1.36.7-2.5 1.7-3.4 2.96-.08.14-.23.21-.39.21zm6.25 12.07c-.13 0-.26-.05-.35-.15-.87-.87-1.34-1.43-2.01-2.64-.69-1.23-1.05-2.73-1.05-4.34 0-2.97 2.54-5.39 5.66-5.39s5.66 2.42 5.66 5.39c0 .28-.22.5-.5.5s-.5-.22-.5-.5c0-2.42-2.09-4.39-4.66-4.39-2.57 0-4.66 1.97-4.66 4.39 0 1.44.32 2.77.93 3.85.64 1.15 1.08 1.64 1.85 2.42.19.2.19.51 0 .71-.11.1-.24.15-.37.15zm7.17-1.85c-1.19 0-2.24-.3-3.1-.89-1.49-1.01-2.38-2.65-2.38-4.39 0-.28.22-.5.5-.5s.5.22.5.5c0 1.41.72 2.74 1.94 3.56.71.48 1.54.71 2.54.71.24 0 .64-.03 1.04-.1.27-.05.53.13.58.41.05.27-.13.53-.41.58-.57.11-1.07.12-1.21.12zM14.91 22c-.04 0-.09-.01-.13-.02-1.59-.44-2.63-1.03-3.72-2.1-1.4-1.39-2.17-3.24-2.17-5.22 0-1.62 1.38-2.94 3.08-2.94 1.7 0 3.08 1.32 3.08 2.94 0 1.07.93 1.94 2.08 1.94s2.08-.87 2.08-1.94c0-3.77-3.25-6.83-7.25-6.83-2.84 0-5.44 1.58-6.61 4.03-.39.81-.59 1.76-.59 2.8 0 .78.07 2.01.67 3.61.1.26-.03.55-.29.64-.26.1-.55-.04-.64-.29-.49-1.31-.73-2.61-.73-3.96 0-1.2.23-2.29.68-3.24 1.33-2.79 4.28-4.6 7.51-4.6 4.55 0 8.25 3.51 8.25 7.83 0 1.62-1.38 2.94-3.08 2.94s-3.08-1.32-3.08-2.94c0-1.07-.93-1.94-2.08-1.94s-2.08.87-2.08 1.94c0 1.71.66 3.31 1.87 4.51.95.94 1.86 1.46 3.27 1.85.27.07.42.35.35.61-.05.23-.26.38-.47.38z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root-180"
            >
              <span
                class="MuiTypography-root-72 MuiTypography-body1-81 MuiListItemText-primary-183"
              >
                Systems
              </span>
            </div>
            <span
              class="MuiTouchRipple-root-187"
            />
          </a>
          <a
            class="MuiButtonBase-root-60 MuiListItem-root-167 MuiListItem-default-170 MuiListItem-gutters-175 MuiListItem-button-176"
            data-testid="navlink-/subscriptions"
            href="/subscriptions"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root-179"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-63"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root-180"
            >
              <span
                class="MuiTypography-root-72 MuiTypography-body1-81 MuiListItemText-primary-183"
              >
                Subscriptions
              </span>
            </div>
            <span
              class="MuiTouchRipple-root-187"
            />
          </a>
        </ul>
      </div>
    </div>
    <main
      class="AppFrame-content-11"
    >
      <div
        class="AppFrame-appBarSpacer-12"
      />
      <div
        class="PageFrame-root-186"
      >
        <div
          data-testid="openapi-layout"
          id="swaggerContainer"
        >
          <div
            class="swagger-ui"
            data-reactroot=""
          >
            <div
              class="loading-container"
            >
              <div
                class="info"
              >
                <div
                  class="loading-container"
                >
                  <div
                    class="loading"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
